rumqttd/src/router/routing.rs

```diff
@@ -1129,15 +1232,52 @@ fn forward_device_data(
         return ConsumeStatus::FilterCaughtup;
     }
 
+    // there are three cases
```

Comment:
Let's take Henil's review on this section for readability

Reply:
@mnpw @henil hey, just did a major refactoring to this, please review the updated code!

Reply:
marking this as resolved, as its outdated.